### PR TITLE
feat(51): Create additional social profile table

### PR DESCRIPTION
**Pull request by hromogen**
Migrate DB to create additional social profile table; 
add possibility to merge user with social details and without by email

### CODE REVIEW

File: packages/server/src/db/db.service.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+import UserEntity from '../models/user.entity';
+import EmailVerificationEntity from '../models/email-verification.entity';
+import PasswordResetEntity from '../models/password-reset.entity';
+import SocialProfileEntity from '../models/social-profile.entity';
+import SocialProfileDto from '../models/social-profile.dto';
+import RegistrationDto from '../models/registration.dto';
+import { ProvidersIdDBFieldName } from 'providers-auth/providers-auth.interfaces';
+
+interface CreateUserAuthEntityPayload {
```
Comment (dartwlad) [inline packages/server/src/db/db.service.ts]: I would suggest moving interfaces to separate files

File: packages/server/src/db/db.service.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+  constructor() {
+    this.userRepository = getRepository(UserEntity);
+    this.emailVerificationRepository = getRepository(EmailVerificationEntity);
+    this.passwordResetRepository = getRepository(PasswordResetEntity);
+    this.socialProfileRepository = getRepository(SocialProfileEntity);
+  }
+
+  public async getUserById(id: string) {
+    return this.userRepository.findOne({ id });
```
Comment (dartwlad) [inline packages/server/src/db/db.service.ts:31]: Shouldn't we use await before DB call?

File: packages/server/src/user-management/user-management.service.ts [hunk 3]
```diff
@@ -60,32 +59,33 @@ export default class UserManagementService {
       throw new EmailResetTokenExpired();
     }
 
-    if (await this.userRepository.findOne({ email })) {
```
Comment (dartwlad) [inline packages/server/src/user-management/user-management.service.ts:63]: I think we should leave current logic.

File: packages/server/src/db/db.service.ts (hunk 2 above)
Comment (hromogen) [reply packages/server/src/db/db.service.ts:31]: It is one call per method so it is going to be `awaited` anyways

Comment (dubbha) [reply packages/server/src/db/db.service.ts:31]: Could you elaborate on this please?

Comment (hromogen) [reply packages/server/src/db/db.service.ts:31]: Well, `this.userRepository.findOne({ id })` returns promise. We do not need this promise result inside the method, it is simply returned, right? So the `user` is not needed to be `awaited` for inside the method. And outside the method we call `await this.dbService.getUserById(...)` meaning the program will anyways wait for the returned promise to be resolved.

Comment (dubbha) [reply packages/server/src/db/db.service.ts:31]: So getUserById method doesn't need to be async I guess.

**Review by oleksiydubovykTR — APPROVED** [review]

**Review by dubbha — APPROVED** [review]

File: packages/server/.eslintignore [hunk 4]
```diff
@@ -1 +1,2 @@
-node_modules
\ No newline at end of file
+node_modules
+migrations
```
Comment (dartwlad) [inline packages/server/.eslintignore]: Previously we decided not to ignore migrations.

**Review by dartwlad — APPROVED** [review]

File: packages/server/migrations/1595162567466-migration.ts [hunk 5]
```diff
@@ -1,22 +1,31 @@
...
-        await queryRunner.query('DROP TABLE "user"');
-        await queryRunner.query('DROP TABLE "password_reset"');
-        await queryRunner.query('DROP TABLE "health_check"');
-        await queryRunner.query('DROP TABLE "email_verification"');
-        await queryRunner.query('DROP TABLE "email_reset"');
-    }
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      'CREATE TABLE "email_reset" ("id" SERIAL NOT NULL, "userId" character varying NOT NULL, "emailResetToken" character varying NOT NULL, "expiresIn" bigint NOT NULL, CONSTRAINT "PK_c7cc1d8ec099867c393d29adafb" PRIMARY KEY ("id"))'
+    );
```
Comment (dubbha) [inline packages/server/migrations/1595162567466-migration.ts:9]: Let's keep these compact, I kind of prefer them like this.
I also just fixed eslint indentation for all packages, this should help avoid fixes related to indentations, let's also merge that and keep that in mind:
https://github.com/dubbha/croton/pull/80

**Pull request merged**